advanced/Scripts/piholeCheckout.sh

```diff
@@ -68,9 +68,20 @@ FTLinstall() {
 }
 
 get_binary_name() {
+  # This gives the machine architecture which may be different from the OS architecture...
   local machine
   machine=$(uname -m)
 
+  # This gives the architecture of packages dpkg installs (for example, "i386")
+  local dpkgarch
+  dpkgarch=$(dpkg --print-architecture 2> /dev/null)
```

Comment:
On fedora, is it ok that dpkg will not exist?

Reply:
dpkg won't be there, but file will be:

```
file $(whereis file) | grep -v "64-bit" | grep "32-bit" &> /dev/null
if [ "$?" = 0 ]; then
  machine="i686"
fi

```
Which would technically work in both debian and fedora universes. But I'd only use it if 1] dpkg wasn't available, and 2] the architecture was identifying as x86_64 via uname -m.